Lib/unittest/mock.py

```diff
@@ -925,10 +926,10 @@ def assert_any_call(self, /, *args, **kwargs):
         The assert passes if the mock has *ever* been called, unlike
         `assert_called_with` and `assert_called_once_with` that only pass if
         the call is the most recent one."""
-        expected = self._call_matcher((args, kwargs))
+        expected = self._call_matcher(_Call((args, kwargs), two=True))
+        cause = expected if isinstance(expected, Exception) else None
         actual = [self._call_matcher(c) for c in self.call_args_list]
-        if expected not in actual:
-            cause = expected if isinstance(expected, Exception) else None
+        if cause or expected not in _AnyComparer(actual):
```

Comment:
One more case: Please do the check for cause to be exception `if cause or expected not in _AnyComparer(actual):` in `assert_any_await` too to catch the exception like you did in `assert_any_call`. This would cause below test to fail. I had a test in my patch for this case. Something like below test :

```
diff --git a/Lib/unittest/test/testmock/testasync.py b/Lib/unittest/test/testmock/testasync.py
index 1b1f9111d2..87f3cfc8fa 100644
--- a/Lib/unittest/test/testmock/testasync.py
+++ b/Lib/unittest/test/testmock/testasync.py
@@ -192,6 +192,10 @@ class AsyncAutospecTest(unittest.TestCase):
         spec.assert_awaited_with(1, 2, c=3)
         spec.assert_awaited()

+        with self.assertRaises(AssertionError):
+            spec.assert_any_await(e=1)
+
+
     def test_patch_with_autospec(self):

         async def test_async():
```

Reply:
Thank you @tirkarthi ! I also saw in your patch you added the `two=True` parameter to `assert_called_with` which strikes me as good for consistency, so I added that back in too.